documentation/modules/ROOT/pages/configuration/topic-auto-create-config.adoc

```diff
@@ -0,0 +1,216 @@
+[cols="1,9",options="header"]
+|===
+|Item |Description
+
+|1
+|First we define the configuration for the `inventory` group.
+
+|2
+|`topic.creation.inventory.include` defines a regular expression to match all topics that start with
+`dbserver1.inventory.`. The config defined for the `inventory` group will only be applied when the
+topic name matches the given regular expression.
+
+|3
+|Then we define the configuration for the `applicationlogs` group.
+
+|4
+|`topic.creation.applicationlogs.include` defines a regular expression to match all topics that start
+with `dbserver1.logs.applog-`. The config defined for the `applicationlogs` group will only be
+applied when the topic name matches the given regular expression. As there's also the `exclude`
+property defined at position *<5>* all topics matching this `include` regular expression might be
+further restricted by the that `exlude` property.
+
+|5
+|`topic.creation.applicationlogs.exclude` defines a regular expression to match all topics that start
+with `dbserver1.logs.applog-old-`. The config defined for the `applicationlogs` group will only be
+applied when the topic name does *not* match the given regular expression. As there's also the
+`include` property set for this group the `applicationlogs` group will only be applied to topics
+which name matches the `include` regular expression/s *and* _not_ match the `exclude` regular
+expression/s.
+|===
+
+=== Registering Custom Groups
+
+Finally, we need to register the two defined custom groups `inventory` and `applicationlogs` with
+the `topic.creation.groups` property:
+
+[source,options="nowrap",json]
+----
+{
+    ...
+
+    "topic.creation.groups": "inventory,applicationlogs",
+
+     ...
+}
+----
+
+A complete connector JSON config will look like that:
+
+[source,options="nowrap",json]
+----
+{
+    ...
+
+    "topic.creation.default.replication.factor": 3,
+    "topic.creation.default.partitions": 10,
+    "topic.creation.default.cleanup.policy": "compact",
+    "topic.creation.default.compression.type": "lz4"
+    "topic.creation.groups": "inventory,applicationlogs",
+    "topic.creation.inventory.include": "dbserver1\\.inventory\\.*",
+    "topic.creation.inventory.replication.factor": 3,
+    "topic.creation.inventory.partitions": 20,
+    "topic.creation.inventory.cleanup.policy": "compact",
+    "topic.creation.inventory.delete.retention.ms": 7776000000,
+    "topic.creation.applicationlogs.include": "dbserver1\\.logs\\.applog-.*",
+    "topic.creation.applicationlogs.exclude": "dbserver1\\.logs\\.applog-old-.*",
+    "topic.creation.applicationlogs.replication.factor": 1,
+    "topic.creation.applicationlogs.partitions": 20,
+    "topic.creation.applicationlogs.cleanup.policy": "delete",
+    "topic.creation.applicationlogs.retention.ms": 7776000000,
+    "topic.creation.applicationlogs.compression.type": "lz4"
+}
+----
+
+ifdef::community[]
+== Additional resources
```

Comment:
An introductory sentence would be good: "To learn more about topic creation settings...".

Reply:
very good point! will do that.

Reply:
Done